src/logging.cc

```diff
@@ -1215,10 +1214,14 @@ void LogMessage::Init(const char* file,
   allocated_ = NULL;
   if (severity != GLOG_FATAL || !exit_on_dfatal) {
 #ifdef GLOG_THREAD_LOCAL_STORAGE
+    const uintptr_t kAlign = sizeof(void*) - 1;
+
     // No need for locking, because this is thread local.
     if (thread_data_available) {
       thread_data_available = false;
-      data_ = new (&thread_msg_data) LogMessageData;
+      char* align_ptr = (char*)(uintptr_t(thread_msg_data + kAlign) & (~kAlign));
```

Comment:
Please don't use C-casts. Use `reinterpret_cast<>` and `static_cast<>` instead for the first and the second cast, respectively.

Reply:
You can not use `static_cast` to convert from pointer to integer. I get the error below. I will use 
twice reinterpret_cast.
```
logging.cc:1233:82: error: invalid static_cast from type ‘char*’ to type ‘uintptr_t {aka long unsigned int}’
           reinterpret_cast<char*>(static_cast<uintptr_t>(thread_msg_data + kAlign) & ~kAlign);
```

Reply:
That's fine. Obviously, this is also good reason for not using C-cast since the expressions become unreadable.

Reply:
Sure